File: eBiblioteka.Servisi/Database/Recenzija.cs
using System;
using System.Collections.Generic;

namespace eBiblioteka.Servisi.Database;

public partial class Recenzija
{
    public int RecenzijaId { get; set; }

    public int? Ocjena { get; set; }

    public string? Opis { get; set; }

    public int? KnjigaId { get; set; }

    public int? KorisnikId { get; set; }

    public bool? Odobrena { get; set; }

    public DateTime DatumRecenzije { get; set; }

    public virtual Knjiga? Knjiga { get; set; }

    public virtual Korisnik? Korisnik { get; set; }
}

File: eBibliotekaUI/ebiblioteka_admin/lib/models/recenzija.dart
import 'package:json_annotation/json_annotation.dart';
import 'knjiga.dart';
import 'korisnik.dart';

part 'recenzija.g.dart';

@JsonSerializable()
class Recenzija {
  int? recenzijaId;
  int? knjigaId;
  int? korisnikId;
  String? opis;
  int? ocjena;
  DateTime? datumRecenzije;
  bool? odobrena;
  Knjiga? knjiga;
  Korisnik? korisnik;

  Recenzija(
      {this.recenzijaId,
      this.knjigaId,
      this.korisnikId,
      this.opis,
      this.ocjena,
      this.datumRecenzije,
      this.odobrena});

  factory Recenzija.fromJson(Map<String, dynamic> json) =>
      _$RecenzijaFromJson(json);

  Map<String, dynamic> toJson() => _$RecenzijaToJson(this);
}
